File: src/algorithms/uncommon-from-sentences.js
/**
 * Given two sentences s1 and s2, returns a list of all the uncommon words.
 * 
 * A word is uncommon if it appears exactly once in one of the sentences, and
 * does not appear in the other sentence.
 * 
 * @param {string} s1 - First string of lowercase letters
 * @param {string} s2 - Second string of lowercase letters
 * @returns {string[]} - Returns an array of uncommon strings
 */
function uncommonFromSentences (s1, s2) {
  let res = [];
  let wordCount = {};
  s1 = s1.split(' ');
  s2 = s2.split(' ');
  for (const word of s1) {
    wordCount[word] = (wordCount[word] || 0) + 1;
  }
  for (const word of s2) {
    wordCount[word] = (wordCount[word] || 0) + 1;
  }
  for (const [key, val] of Object.entries(wordCount)) {
    if (val === 1) {
      res.push(key);
    }
  }
  return res;
}

File: src/algorithms/uncommon-from-sentences.py
def uncommon_from_sentences(s1, s2):
    """A sentence is a string of single-space separated words where each word 
    consists only of lowercase letters. A word is uncommon if it appears 
    exactly once in one of the sentences, and does not appear in the other 
    sentence. Given two sentences s1 and s2, return a list of all the uncommon 
    words. You may return the answer in any order.

    Args:
        s1 (str): A string of lowercase English letters and spaces.
        s2 (str): A string of lowercase English letters and spaces.

    Returns:
        list[str]: Returns a list of unique words found in s1 and s2.s
    """
    res = []
    seen = {}
    s1 = s1.split()
    s2 = s2.split()
    for word in s1:
        if word in seen:
            seen[word] += 1
        else:
            seen[word] = 1
    for word in s2:
        if word in seen:
            seen[word] += 1
        else:
            seen[word] = 1
    for key, val in seen.items():
        if val == 1:
            res.append(key)
    return res
